tests/vue/multiparser/jsfmt.spec.js

```diff
@@ -1 +1,18 @@
-run_spec(__dirname, ["vue"]);
+run_spec({dirname: __dirname, snippets: [
+  {
+    name: "empty",
+    code: '<custom lang="markdown"></custom>'
+  },
+  {
+    name: "spaces",
+    code: '<custom lang="markdown">   </custom>'
+  },
+  {
+    name: "new line",
+    code: '<custom lang="markdown">\n \n</custom>'
+  },
+  {
+    name: "non-space",
+    code: '<custom lang="markdown">\n \u2005 \n</custom>'
```

Comment:
This one, html think it's not empty, and `markdown` will print to a empty `Doc`, how do I tell it's empty?

Reply:
The only child node is text consisting of whitespace only.

Reply:
```json
{
      "type": "concat",
      "parts": [
        {
          "type": "concat",
          "parts": [
            {
              "type": "fill",
              "parts": []
            }
          ]
        }
      ]
    }
```

Reply:
Try using `normalizeParts` from `doc-utils`. You might want to improve it to handle an empty `fill` doc.

Reply:
What if the embed parser return `ifBreak('foo', '')`, we should not be able to know it's empty or not. Am I right?

Reply:
Can't we first check if the content of the tag is empty or whitespace-only and skip the embedded parser if it is?

Reply:
I tried, and it works, just feel something not right.

Reply:
It's fine to assume that reformatting an empty document should result in an empty document for any language.